### DISCUSSION TITLE
get the current wallpaper path?

### DISCUSSION THREAD

**Discussion by H1manshu-Soni** [prasanthrangan/hyprdots#1055] [Q&A] [answered]
I personally like to use pywal to theme firefox according to the wallpaper ( its magical )
Now, the workaround that I had, to generate color-schemes has stopped working. if anyone knows what might be the issue your answer will be really helpful, I will attach the part of the ``` swwwallpaper.sh ``` script that I have changed.

```
Wall_Set()
{
    if [ -z $xtrans ] ; then
        xtrans="grow"
    fi

    #? getting the real path as symlinks too glitch
    swww img "$(readlink "${wallSet}")" \
    --transition-bezier .43,1.19,1,.4 \
    --transition-type "$xtrans" \
    --transition-duration 0.7 \
    --transition-fps 60 \
    --invert-y \
    --transition-pos "$( hyprctl cursorpos )"    
    
    wal -c
    wal -i ${wallSet}
    pywalfox update

}
```

Also I thought of creating a small script that will generate color-schemes based on wallpaper and run it every time the wallpaper changes but I couldn't figure out a way of getting the current wallpaper path.

P.S. : I use my own set of wallpapers but I have distributed them accordingly in their supposed themes folder in ``` .config/swww/<THEME_NAME> ``` but that should not pose any problems ( from my understanding )

**Comment by kRHYME7** [comment]
The most optimal way is understanding how this works.
https://github.com/prasanthrangan/hyprdots/wiki/Wallbash

Gotta help you out later as I'm not home.

You just need 2 files to make this work and also will be modular and isolated to changes.

I can't make pywalfox work so I have a hard time helping.

**Reply by H1manshu-Soni** [reply]
As I am sure you would have understood by my workaround I dont have very good knowledge about this stuff I just know small bits, which I implement ( most of the times rather _poorly_  ) although I will give it a re-read and see if I can fix it by myself.
I would really like to see your solution with the 2 files sounds interesting.

**Reply by kRHYME7** [reply]
can you provide me your script that handles the part of pywalfox? Just leave the wallpaper variable blank.

**Reply by H1manshu-Soni** [reply]
I dont have a script I just use these three lines in the ``` swwwallpaper.sh ```  in the wallSet function.
```
    wal -c
    wal -i ${wallSet}
    pywalfox update
 ```
 
 {wallSet} was supposed to be the wallpaper variable

**Reply by kRHYME7** [reply, marked as answer]
Hmm, okay I realized now that you're using the older script directory. 

This is the new script directory, so you can add your own config here 

``` ~/.config/hyprdots/scripts/swwwallpaper.sh ``` 

it's now in ``` ~./config/hyprodts/scripts/ ```

Try.

**Comment by kRHYME7** [comment]
Now we got gtk wallbash template so any gtk apps will follow the wallpaper color scheme. 

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/a909ff06-3f48-4165-8024-bb268cb1e7de)

**Reply by H1manshu-Soni** [reply]
I pulled the latest changes and did a re-install but firefox didnt get themed according to the wallpaper.

**Reply by kRHYME7** [reply]
how did you update? You enabled Wallbash?

**Reply by H1manshu-Soni** [reply]
git pull
./install.sh 

wallbash mode: theme mode

**Reply by kRHYME7** [reply]
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/6034f6c7-6804-4272-9ee2-e7e7debd3696)
` ls ~/.config/hyde/wallbash/Wall-Dcol/gtk ` 
check this files first

**Reply by H1manshu-Soni** [reply]
yes i have the same files

**Reply by kRHYME7** [reply]
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/a2f1a132-bf9b-4402-a3b9-f9d4aad8b22a)

` swwwallselect.sh `

**Reply by H1manshu-Soni** [reply]
![image](https://github.com/prasanthrangan/hyprdots/assets/84933847/3d058a05-57af-4a33-b1f6-0599bc117f9b)

**Reply by kRHYME7** [reply]
The error is fine it's just the older ~/.vscode

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/9eb8bd72-aea8-4b5f-a8f0-2d61896a661f)


everything looks normal. Can you check some gtk apps? with screenshots.

**Reply by kRHYME7** [reply]
You should also have this directory in ~/.themes 
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/5c6529f6-b19a-4e3a-b169-c2efa4e20471)

` ~/.themes/Wallbash-Gtk/ `

**Reply by H1manshu-Soni** [reply]
I changed my wallbash from theme to auto and it workd now : )
Thanks a lot for implementing this and also for letting me know about this even after so long.

Love this project. :100:
